src/runtime/rpc/asio_net_provider.h

```diff
@@ -45,13 +45,16 @@ class asio_network_provider : public connection_oriented_network
 
 private:
     void do_accept();
+    boost::asio::io_service &get_io_service();
 
 private:
     friend class asio_rpc_session;
     friend class asio_network_provider_test;
 
     std::shared_ptr<boost::asio::ip::tcp::acceptor> _acceptor;
-    boost::asio::io_service _io_service;
+    int _next_io_service = 0;
+    typedef std::shared_ptr<boost::asio::io_service> io_service_ptr;
```

Comment:
In my opinion, it's better to use std::unique_ptr here

Reply:
Maybe raw point is better